### PR TITLE
Jwk part one

### PR DESCRIPTION
**Pull request by Erik1000**

### CODE REVIEW

File: src/jwk/rsa.rs [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+}
+
+impl<'de> Deserialize<'de> for RsaPrivateKey {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: serde::Deserializer<'de>,
+    {
+        #[derive(Deserialize)]
+        struct Repr {
+            kty: String,
```
Comment (Erik1000) [inline src/jwk/rsa.rs:139]: does this have to be a String?

File: src/format.rs [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+    fn into_format(self) -> F;
+}
+
+/// Internal trait.
+#[doc(hidden)]
+pub trait AppendSignatureTo<F>: sealed::Sealed {
+    fn append_to(self, f: &mut F);
+}
+
+impl<T: AsRef<[u8]>> sealed::Sealed for T {}
```
Comment (Erik1000) [inline src/format.rs:44]: Doesn't this implement `Sealed` for every `T: AsRef<[u8]>` including types outside this crate?

**Review by Stupremee — APPROVED** [review]

**Review by Stupremee — APPROVED** [review]

**Comment by Stupremee** [comment]
bors merge

**Comment by Stupremee** [comment]
bors retry

**Comment by Stupremee** [comment]
bors merge

**Comment by Erik1000** [comment]
bors ping

**Comment by Erik1000** [comment]
bors merge

**Comment by Erik1000** [comment]
bors cancel

**Pull request merged**